packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -3447,6 +3446,8 @@ export function attach(
     const stashedPrevious = previouslyReconciledSibling;
     const stashedRemaining = remainingReconcilingChildren;
     if (fiberInstance !== null) {
+      // Update the Fiber so we that we always keep the current Fiber on the data.
+      fiberInstance.data = nextFiber;
```

Comment:
This revealed this bug. I wasn't updating the fiber in all places that I called updateFiberRecursively(). It should really be in here for parity with the create path.